Sources/OpenAPIRuntime/Multipart/MultipartFramesToBytesSequence.swift

```diff
@@ -0,0 +1,71 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the SwiftOpenAPIGenerator open source project
+//
+// Copyright (c) 2023 Apple Inc. and the SwiftOpenAPIGenerator project authors
+// Licensed under Apache License v2.0
+//
+// See LICENSE.txt for license information
+// See CONTRIBUTORS.txt for the list of SwiftOpenAPIGenerator project authors
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+//===----------------------------------------------------------------------===//
+
+import HTTPTypes
+
+/// A sequence that serializes multipart frames into bytes.
+struct MultipartFramesToBytesSequence<Upstream: AsyncSequence & Sendable>: Sendable
+where Upstream.Element == MultipartFrame {
+
+    /// The source of multipart frames.
+    var upstream: Upstream
+
+    /// The boundary string used to separate multipart parts.
+    var boundary: String
+}
+
+extension MultipartFramesToBytesSequence: AsyncSequence {
+
+    /// The type of element produced by this asynchronous sequence.
+    typealias Element = ArraySlice<UInt8>
```

Comment:
How do you feel about making use of the byte chunk here.

```suggestion
    typealias Element = HTTPBody.ByteChunk
```

Reply:
Yeah I had that originally, but then I found it nicer in tests to not couple it to HTTPBody in any way.